### PR TITLE
167 unify cards

### LINKED ISSUE #167: Unify the view cards
### Description
currently we use four different view cards 

- [ ]  unify all the card into one to two classes
- [ ]  create a class for all styling so it's easily modifiable 

### Expected Results
- [ ]  cleaner code

### PR DESCRIPTION
**Pull request by Sainaamr**
Corresponding Issue:

 resolves #167

What was done:

Download card, pinned card and stream unified under one class and then break down into three using inheritance to avoid repetitive code. The look of the cards has not been changed.

### CODE REVIEW

**Review by Anishyou — CHANGES_REQUESTED** [review]
I would suggest you put Pinned_Card under pinned_courses_view, similar with others. Big_card can be in components. 
Also change name of Big_Card to something else to understand better. e.g. just card_view

File: lib/views/course_view/components/big_card.dart
Comment (GravityDarkLab) [inline lib/views/course_view/components/big_card.dart:1]: Looks good.
But maybe change the name to just `Card` also file name to `card.dart`

**Review by GravityDarkLab — CHANGES_REQUESTED** [review]
FIle naming changes for better understanding:

`big_card.dart --> card.dart`
`course_card_view.dart --> course_card.dart`

Also place pinned under pinned, download under download and so on...

File: lib/views/course_view/components/pinned_card.dart [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    required this.course,
+    required this.isPinned,
+    required this.onPinToggle,
+  });
+
+  @override
+  List<Widget> buildCardContent() {
+    return [
+      buildHeader(
+        title: '${course.name} - ${course.slug}',
```
Comment (GravityDarkLab) [inline lib/views/course_view/components/pinned_card.dart:23]:
```suggestion
        title: '${course.name} - ${course.tUMOnlineIdentifier}',
```

**Review by GravityDarkLab — CHANGES_REQUESTED** [review]

**Review by GravityDarkLab — DISMISSED** [review]
Looks good to me

**Review by Anishyou — APPROVED** [review]

**Review by GravityDarkLab — APPROVED** [review]

**Pull request merged**